altair/utils/core.py

```diff
@@ -106,6 +107,9 @@ def to_list_if_array(val):
         else:
             return val
 
+    # All column names must be strings
+    df.columns = map(str, df.columns)
```

Comment:
If we want to convert column names to strings, we should probably put this line in a `try/expect` that raises an informative error telling the user how to fix the issue.